src/Workspaces/CSharp/Portable/CodeGeneration/ExpressionGenerator.cs

```diff
@@ -311,5 +312,57 @@ private static ExpressionSyntax GenerateMemberAccess(params string[] names)
             result = result.WithAdditionalAnnotations(Simplifier.Annotation);
             return result;
         }
+
+        private static ExpressionSyntax GenerateSByteLiteralExpression(ITypeSymbol type, sbyte value, bool canUseFieldReference)
+        {
+            if (canUseFieldReference || value > sbyte.MinValue)
+            {
+                return GenerateLiteralExpression(type, value, LiteralSpecialValues.SByteSpecialValues, null, canUseFieldReference, (s, v) => SyntaxFactory.Literal(s, v), x => x < 0, x => (sbyte)-x);
+            }
+
+            // We have to special case how sbyte.MinValue is printed when we can't refer to the
+            // field directly.
+            var numericLiteralToken = SyntaxFactory.Literal("-128", value);
```

Comment:
this is incorrect.  we should be generating a NegateExpression here around the LiteralExpression for 128.

Reply:
Updated